source/iface/native-prec-math-func.rst

```diff
@@ -7,3 +7,407 @@
 *******************************
 Native precision math functions
 *******************************
+
+In SYCL the implementation-defined precision math functions are defined in the
+namespace ``sycl::native``. The functions that are available within this
+namespace are specified in the list below.
+
+The range of valid input values and the maximum error for these functions
+is implementation defined.
+
+``cos``
+=======
+
+::
+
+  float cos(float x)                (1)
+
+  template<typename NonScalar>      (2)
+  /*return-type*/ cos(NonScalar x)
+
+Overload (1):
+
+Returns: The cosine of ``x``.
+
+Overload (2):
+
+Constraints: Available only if all of the following conditions are met:
+
+NonScalar is ``marray``, ``vec``, or the ``__swizzled_vec__`` type; and
+
+The element type is ``float``.
```

Comment:
In all the entries here, I think the conditions (second and third lines in this example) should be in list form.  It isn't clear which are the constraints otherwise.

Reply:
Fixed with : https://github.com/KhronosGroup/SYCL_Reference/pull/165/commits/bd4ad605d392d264e4bf0e71457a39513a7133b6